### PR TITLE
Add SPNEGO authentication header

### LINKED ISSUE #106: Add SPNEGO auth to response headers
Add SPNEGO header so users are more easily aware that it is an option.
Currently, the endpoint only advertises Basic Auth.
spnego-not-correctly-advertized-in-ls 1

### PR DESCRIPTION
**Pull request by TheLydonKing**
Added WWW-Authentication: Negotiate Header to 401 Status response when Kerberos is enabled

closes #106

### CODE REVIEW

File: api/src/main/scala/za/co/absa/loginsvc/rest/SecurityConfig.scala [hunk 1]
```diff
@@ -58,13 +61,20 @@ class SecurityConfig @Autowired()(authConfigsProvider: AuthConfigProvider) {
...
+          http.addFilterBefore(
               kerberos.spnegoAuthenticationProcessingFilter,
               classOf[BasicAuthenticationFilter])
-          }
+              .exceptionHandling()
+              .authenticationEntryPoint((request: HttpServletRequest,
+                                         response: HttpServletResponse,
+                                         authException: AuthenticationException) => {
+                response.setStatus(HttpServletResponse.SC_UNAUTHORIZED)
+                response.addHeader("WWW-Authenticate", "Negotiate")
```
Comment (dk1844) [inline api/src/main/scala/za/co/absa/loginsvc/rest/SecurityConfig.scala:75]: I tried it and when kerberos was disabled in config, running a POST against the endpoint ( `curl -X POST http://localhost:9090/token/generate -v`) resulted only in:
`< WWW-Authenticate: Basic realm="Realm"`
(as expected ✅ ).

However, when running the same while having kerberos enabled in config, it only returned
`< WWW-Authenticate: Negotiate`
This time I would expect both options being present in the headers:
`< WWW-Authenticate: Basic realm="Realm"`
`< WWW-Authenticate: Negotiate`

**Review by dk1844 — COMMENTED** [review]
There is also the matter of existing _Swagger v3 OAS Annotations_ (aka OpenAPI docs) - this has bears no information about the Negotiate Auth option.
`GET http://localhost:9090/v3/api-docs.yaml` gives
```
  /token/generate:
    post:
      security:
      - basicAuth: []
  ...
  securitySchemes:
    basicAuth:
      type: http
      scheme: basic
...
```

but it should probably give
```
  /token/generate:
    post:
      security:
      - negotiate: []
      - basicAuth: []
  ...
  securitySchemes:
    negotiate:
      type: http
      scheme: negotiate
    basicAuth:
      type: http
      scheme: basic
```

I think that the solution *might* be adding another
```scala
  @SecurityRequirement(name = "negotiate")
```
and introducing something like
```scala
@SecurityScheme(
  name = "negotiate",
  `type` = SecuritySchemeType.HTTP,
  scheme = "negotiate"
)

I am not sure that it is entirely correct (perhaps do your own research how it should be done properly, but something along these lines

File: api/src/main/scala/za/co/absa/loginsvc/rest/SecurityConfig.scala (hunk 1 above)
Comment (TheLydonKing) [reply api/src/main/scala/za/co/absa/loginsvc/rest/SecurityConfig.scala:75]: Hi Daniel

I see the issue. I was testing and accidentally left -u in the curl command and that would get WWW-Authenticate: Basic realm="Realm" back along with the Negotiate. So it looked like it was working properly on my end. Will sort this out along with the OpenAPI.

**Review by dk1844 — APPROVED** [review]
LGTM (tested on my machine)

**Pull request merged**